gapic/src/main/com/google/gapid/views/AtomTree.java

```diff
@@ -206,47 +212,67 @@ public void widgetSelected(SelectionEvent e) {
       @Override
       public void mouseDown(MouseEvent e) {
         Point location = new Point(e.x, e.y);
-        Object follow = /*TODO(CanFollow)*/labelProvider.getFollow(location);
+        Path.Any follow = (Path.Any)labelProvider.getFollow(location);
         if (follow != null) {
-          //models.follower.follow(getFollowPath(tree.getItem(location), follow.getPath()));
+          models.follower.onFollow(follow);
         }
       }
 
+      @Override
+      public void mouseExit(MouseEvent e) {
+        hoverItem(null);
+        hoverImage(null);
+      }
+
       private void updateHover(int x, int y) {
         TreeItem item = tree.getItem(new Point(x, y));
-        if (item != null && (item.getData() instanceof AtomStream.Node) &&
-            item.getImage() != null && item.getImageBounds(0).contains(x, y)) {
-          hover(item);
-        } else {
-          hover(null);
+        // When hovering over the far left of deep items, getItem returns null. Let's check a few
```

Comment:
Yeah... and it's not actually SWT's fault either. On OSX it also appears to be inconsistent :(